manifests/Makefile

```
@@ -33,34 +45,47 @@ define BUILD_BASE
 	fi
 endef
 
+define BUILD_TEMPLATE
+	@if [ -d "upstream/$*/template" ]; then \
+		echo "Generating $@..." && \
+		out=$$($(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone upstream/$*/template) && \
+		printf "# This file was generated by running 'make $@' in manifests directory\n%s" "$$out" > $@; \
+	fi
+endef
+
 
 .SECONDEXPANSION:
-crd/%.yaml: $$(wildcard upstream/$$*/crd/*) $$(wildcard upstream/$$*/helm/*.yaml) bin/kustomize
+crd/%.yaml: $$(wildcard upstream/$$*/crd/*) $$(wildcard upstream/$$*/helm/*.yaml) .bin/kustomize .bin/helm .bin/yq
 	$(BUILD_HELM_CHART)
 	$(BUILD_CRD)
 
 .SECONDEXPANSION:
-%.yaml: $$(wildcard upstream/$$*/base/*) $$(wildcard upstream/$$*/helm/*.yaml) bin/kustomize
+%.yaml: $$(wildcard upstream/$$*/base/*) $$(wildcard upstream/$$*/helm/*.yaml) .bin/kustomize .bin/helm .bin/yq
 	$(BUILD_HELM_CHART)
 	$(BUILD_BASE)
 
-.PHONY: build build-upstream-helm-charts build-crds build-base
+.SECONDEXPANSION:
+template/%.yaml.raw: $$(wildcard upstream/$$*/template/*) $$(wildcard upstream/$$*/helm/*.yaml) .bin/kustomize .bin/helm .bin/yq
+	$(BUILD_HELM_CHART)
+	$(BUILD_TEMPLATE)
+
+.PHONY: build build-upstream-helm-charts build-crds build-base build-templates build-manifests
 
 OS   = $(shell uname -s | tr '[:upper:]' '[:lower:]')
 ARCH = $(shell uname -m | sed 's/x86_64/amd64/')
 
-bin/kustomize:
-	@mkdir -p bin
-	@curl -sSLo - https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize/v4.0.3/kustomize_v4.0.3_$(OS)_$(ARCH).tar.gz | tar xzf - -C bin/ && chmod +x bin/kustomize
+.bin/kustomize:
+	@mkdir -p .bin
+	@curl -sSLo - https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize/v4.0.3/kustomize_v4.0.3_$(OS)_$(ARCH).tar.gz | tar xzf - -C .bin/ && chmod +x .bin/kustomize
 
-bin/yq:
-	@mkdir -p bin
-	@curl -sSLo bin/yq https://github.com/mikefarah/yq/releases/download/3.4.1/yq_$(OS)_$(ARCH) && chmod +x bin/yq
+.bin/yq:
+	@mkdir -p .bin
+	@curl -sSLo .bin/yq https://github.com/mikefarah/yq/releases/download/3.4.1/yq_$(OS)_$(ARCH) && chmod +x .bin/yq
 
-bin/helm:
-	@mkdir -p bin
-	@curl -sSLo - https://get.helm.sh/helm-v3.5.2-$(OS)-$(ARCH).tar.gz | tar xzf - -C bin/ --strip-components=1 '*/helm' && chmod +x bin/helm
+.bin/helm:
+	@mkdir -p .bin
+	@curl -sSLo - https://get.helm.sh/helm-v3.5.2-$(OS)-$(ARCH).tar.gz | tar xzf - -C .bin/ --strip-components=1 '*/helm' && chmod +x .bin/helm
```

This will make sure this `tar` command works for both GNU Tar (Linux) and bsdtar (MacOS). I tested on both platforms.

```suggestion
	@curl -sSLo - https://get.helm.sh/helm-v3.5.2-$(OS)-$(ARCH).tar.gz | tar xzf - -C .bin/ --strip-components=1 $$(tar --version | grep -q 'gnu' && echo "--wildcards") '*/helm' && chmod +x .bin/helm
```